ui-tests/src/test/java/io/syndesis/qe/steps/customizations/extensions/TechnicalExtensionSteps.java

```diff
@@ -1,62 +1,89 @@
 package io.syndesis.qe.steps.customizations.extensions;
 
+import static com.codeborne.selenide.Condition.visible;
+import static com.codeborne.selenide.Selenide.$;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import org.openqa.selenium.By;
+
 import cucumber.api.PendingException;
 import cucumber.api.java.en.Then;
 import cucumber.api.java.en.When;
+import io.syndesis.qe.TestConfiguration;
+import io.syndesis.qe.pages.ModalDialogPage;
+import io.syndesis.qe.pages.customizations.CustomizationsPage;
+import io.syndesis.qe.pages.customizations.extensions.TechExtensionsImportPage;
+import lombok.extern.slf4j.Slf4j;
 
+@Slf4j
 public class TechnicalExtensionSteps {
+	
+	private CustomizationsPage customizationsPage = new CustomizationsPage();
+	private TechExtensionsImportPage techExtensionsImportPage = new TechExtensionsImportPage();
+
+	@Then("^she is presented with dialog page \"([^\"]*)\"$")
+	public void isPresentedWithDialogPage(String title) throws Throwable {
+		String titleText = new ModalDialogPage().getTitleText();
+		assertThat(titleText.equals(title), is(true));
+	}
+
+	@When("^she clicks on the modal dialog \"([^\"]*)\" button$")
+	public void clickModalButton(String buttonName) throws Exception {
+		new ModalDialogPage().getButton(buttonName).shouldBe(visible).click();;
+    	
+	}
+
+	@When("^Camilla upload extension$")
+	public void uploadFile() throws Throwable {
+		//TODO temporary solution
+		String techExtensionUrl = TestConfiguration.techExtensionUrl();
```

Comment:
It it temporary because of the bug? Other then uploading same extension twice?

Reply:
I added temporary comment to not to forget about different future inclusion of extension jar.